joomla/joomla-cms
# [4.0] Namespace com_search

Pull Request for Issue # .

### Summary of Changes
This simple PR converts com_search component to namespace MVC for Joomla 4.

### Testing Instructions
1. Install latest Joomla 4.0-dev with sample data, then apply patch (or better install my branch https://github.com/joomdonation/joomla-cms/archive/namespace_com_search.zip )

2. Login to administrator area, access to Components -> Search, click on Options button in the toobar, set **Gather Search Statistics** parameter to **Yes**

3. Access to frontend of the site, try to search for something using search module, make sure the search result is displayed.

4. Access to administrator area, access to Components -> Search again, make sure the search phrases which you searched before is displayed.

### Additional comment
Right now, we still have to use **\JLoader::register('SearchHelper', JPATH_ADMINISTRATOR . '/components/com_search/helpers/search.php');** in Controller display method because plugins are using SearchHelper class without registering for it. Later, we can modify the search plugin to use namespace helper class directly and if it is allowed , we can remove the SearchHelper class in later phase
